Dockerfile

```diff
@@ -27,20 +27,35 @@ RUN apt-get update && \
     apt-get autoclean -y && \
     rm -rf /var/lib/apt/lists/*
 
+# --------------------------------------
 # Install TeX Live
 # ref: https://tug.org/texlive/quickinstall.html, https://github.com/Paperist/texlive-ja/blob/main/debian/Dockerfile
+# --------------------------------------
+
+# Set ARG and ENV for installation of TeX Live
+# `TEXLIVE_VERSION` is used on `docker-entrypoint.sh`
+ARG ARCHIVE_URL="https://mirror.ctan.org/systems/texlive/tlnet/install-tl-unx.tar.gz"
+ENV TEXLIVE_VERSION "2023"
+
+# Install TeX Live
 WORKDIR /tmp/install-tl-unx
 COPY ./texlive.profile ./
 RUN wget -nv ${ARCHIVE_URL} && \
   tar -xzf ./install-tl-unx.tar.gz --strip-components=1 && \
-  ./install-tl --profile=./texlive.profile && \
+  ./install-tl --profile=./texlive.profile --no-interaction && \
   rm -rf /tmp/install-tl-unx
 
-# Add TeX Live to PATH
-ENV PATH /usr/local/texlive/${TEXLIVE_VERSION}/bin/x86_64-linux:$PATH
+# --------------------------------------
+# Install LaTeX packages
+# --------------------------------------
+
+# Copy `docker-entrypoint.sh` for adding TeX Live binaries to PATH
+COPY ./docker-entrypoint.sh /
+RUN chmod +x /docker-entrypoint.sh
 
 # Install LaTeX packages with tlmgr
-RUN tlmgr update --self --all && tlmgr install \
+RUN . /docker-entrypoint.sh && \
+  tlmgr update --self --all && tlmgr install \
```

Comment:
In `docker-entrypoint.sh`, add TeX Live binaries to `PATH` dynamically using a command `$(uname -m)`.

To use `tlmgr`, we must execute `docker-entrypoint.sh`. This is because `Dockerfile` doesn't support dynamic command when setting `ENV` like `ENV ARCH $(uname -m)`.